src/core/src/local_loggers/CompositeLogger.py

```diff
@@ -77,3 +91,9 @@ def __remove_substring_from_message(message, substring=Constants.ERROR_ADDED_TO_
             message = message.replace("[{0}]".format(Constants.ERROR_ADDED_TO_STATUS), "")
         return message
 
+    def __add_telemetry_error_note(self, message):
+        """ Adding telemetry error messages to the original message, so as to log telemetry errors in other logs """
+        if len(message.encode('utf-8')) > Constants.TELEMETRY_MSG_SIZE_LIMIT_IN_BYTES:
+            message = message + self.telemetry_msg_truncated_note
+        return message
```

Comment:
Composite logger is holding knowledge/responsibility of telemetry it shouldn't have and creating a stronger coupling between the two classes. Telemetry should report such an issue back to composite logger using the aforementioned one-off log level for telemetry.